### PR TITLE
Zero init for common types

### LINKED ISSUE #1494: Support for zero initializing values in CProver
https://github.com/model-checking/kani/pull/1230

### PR DESCRIPTION
**Pull request by danielsn**
### Description of changes: 

Currently, we have support for zero-initilizing bit-vector types in cprover, but not aggregate types.
This PR adds support for aggregate types also.

### Resolved issues:

Resolves #1494 

### Call-outs:
This is phase 1 of a two phase process to add support for zero-init to Kani.
This was split from #1230 to allow for a second PR to address the need to guard init with a config.

### Testing:

* How is this change tested? Using the tests in #1230

* Is this a refactor change? No

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: cprover_bindings/src/goto_program/expr.rs [hunk 1]
```diff
@@ -869,8 +884,9 @@ impl Expr {
         symbol_table: &SymbolTable,
     ) -> Self {
         let field = field.into();
-        assert!(typ.is_union_tag());
+        assert!(typ.is_union_tag() || typ.is_union());
         assert_eq!(typ.lookup_field_type(field, symbol_table).as_ref(), Some(value.typ()));
+        let typ = typ.aggr_tag().unwrap();
```
Comment (celinval) [inline cprover_bindings/src/goto_program/expr.rs:889]: nit: this is inconsistent with the change to `struct_expr_from_padded_values`:
```
let typ = if typ.is_struct() { typ.aggr_tag().unwrap() } else { typ };
```

File: cprover_bindings/src/goto_program/expr.rs [hunk 2]
```diff
@@ -553,6 +562,11 @@ impl Expr {
         Self::float_constant(c)
     }
 
+    /// `typ x[__CPROVER_infinity()] = >>> {elem} <<<`
```
Comment (celinval) [inline cprover_bindings/src/goto_program/expr.rs:565]: Can you please improve this doc a bit? I have no idea what this means.

**Review by celinval — APPROVED** [review]
Just some minor comments

File: cprover_bindings/src/goto_program/expr.rs (hunk 2 above)
Comment (danielsn) [reply cprover_bindings/src/goto_program/expr.rs:565]: Done

**Pull request merged**